File: src/domain/Collection.ts
// ─── 合集/系列领域类型 ───
// 纯数据定义，不含业务逻辑

export interface Article {
  id: string;
  title: string;
  createdAt: number;
  updatedAt: number;
  pinned?: boolean;
  description?: string;
  tags?: string[];
  phase?: string;
  blueprint?: string;
}

export interface Collection {
  id: string;
  title: string;
  description?: string;
  coverImage?: string;
  articles: Article[];
  createdAt: number;
  linkedFolder?: string;
}

export interface SeriesArticle {
  id: string;
  title: string;
  description: string;
  targetWordCount?: number;
  status: "planned" | "outlining" | "writing" | "reviewing" | "complete" | "draft" | "published";
  articleId?: string;
  previousArticleId?: string;
  nextArticleId?: string;
  order?: number;
}

export interface SeriesPlan {
  id: string;
  title: string;
  createdAt: number;
  tone?: string;
  targetAudience?: string;
  skillId?: string;
  styleId?: string;
  actionId?: string;
  articles: SeriesArticle[];
  description?: string;
  totalArticles?: number;
  updatedAt?: number;
}

export interface TrashItem {
  id: string;
  title: string;
  collectionId: string;
  collectionTitle: string;
  deletedAt: number;
  articleId?: string;
  originalCollectionId?: string;
  phase?: string;
}

export interface SearchResult {
  articleId: string;
  collectionId: string;
  collectionTitle: string;
  title: string;
  matchType: "title" | "content";
  snippet?: string;
  score: number;
}

File: src-tauri/src/domain/collection.rs
// ─── Collection / Series 领域类型 ───

use serde::{Deserialize, Serialize};

use super::ArticleMeta;

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct Collection {
    pub id: String,
    pub title: String,
    pub created_at: u64,
    pub linked_folder: Option<String>,
    pub articles: Vec<ArticleMeta>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct SeriesPlan {
    pub id: String,
    pub title: String,
    pub skill_id: Option<String>,
    pub style_id: Option<String>,
    pub action_id: Option<String>,
    pub tone: Option<String>,
    pub target_audience: Option<String>,
    pub created_at: u64,
    pub articles: Vec<SeriesArticle>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct SeriesArticle {
    pub id: String,
    pub title: String,
    pub description: String,
    pub target_word_count: Option<u32>,
    pub status: String,
    pub article_id: Option<String>,
    pub previous_article_id: Option<String>,
    pub next_article_id: Option<String>,
}
